tests/Unit/Utilities/Test_Algorithm.cpp

```diff
@@ -0,0 +1,57 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <deque>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/Algorithm.hpp"
+
+SPECTRE_TEST_CASE("Unit.Utilities.indices_of", "[Unit][Utilities]") {
+  const auto helper = [](const auto& data) {
+    CHECK(std::array<size_t, 3>{{0, 5, 3}} ==
+          indices_of<3>(data, [](const double lhs, const double rhs) {
```

Comment:
If the order of the returned indices is unspecified, do we really want to be testing that the function returns them in a specific order?

Reply:
I guess technically the order is specified, but not guaranteed in the interface because it would be "nice" to swap out the implementation. I'm okay with changing it to finding the values, it would make the test longer, but I'm happy to do it if you think that's the right thing :)

Reply:
I'm not sure what the distinction you are drawing between "specified" and "guaranteed in the interface" is.  Specified behavior is behavior that is not allowed to change if you swap out the implementation.

Anyway, I believe wrapping the calls in `[](const auto& c) noexcept { return std::unordered_set<size_t>(c.begin(), c.end()); }` (or as an actual function) would be a concise solution to the problem.